src/inspector_agent.cc

```diff
@@ -706,6 +791,77 @@ void Url(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(OneByteString(env->isolate(), url.c_str()));
 }
 
+static void InvokeAsyncTaskFnWithId(void (Agent::*asyncTaskFn)(void*),
+                                    const FunctionCallbackInfo<Value>& args);
+
+static void* GetAsyncTask(int64_t asyncId);
+
+static void AsyncTaskScheduledWrapper(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  Local<String> taskName = args[0].As<String>();
+  String::Value taskNameValue(taskName);
+  StringView taskNameView(*taskNameValue, taskNameValue.length());
+
+  CHECK(args[1]->IsNumber());
+  int64_t task_id = args[1]->IntegerValue(env->context()).FromJust();
+  void* task = GetAsyncTask(task_id);
+
+  CHECK(args[2]->IsBoolean());
+  bool recurring = args[2]->BooleanValue(env->context()).FromJust();
+
+  env->inspector_agent()->AsyncTaskScheduled(taskNameView, task, recurring);
+}
+
+static void AsyncTaskCanceledWrapper(const FunctionCallbackInfo<Value>& args) {
+  InvokeAsyncTaskFnWithId(&Agent::AsyncTaskCanceled, args);
+}
+
+static void AsyncTaskStartedWrapper(const FunctionCallbackInfo<Value>& args) {
+  InvokeAsyncTaskFnWithId(&Agent::AsyncTaskStarted, args);
+}
+
+static void AsyncTaskFinishedWrapper(const FunctionCallbackInfo<Value>& args) {
+  InvokeAsyncTaskFnWithId(&Agent::AsyncTaskFinished, args);
+}
```

Comment:
I was thinking about defining `InvokeAsyncTaskFnWithId` as:

```c++
template <void (Agent::*func)(void*)>
static void InvokeAsyncTaskFnWithId(const FunctionCallbackInfo<Value>& args) {
  // ...
  (env->inspector_agent()->*func)(GetAsyncTask(task_id));
}
```

and when a function pointer is needed do something like

```c++
env->SetMethod(target, "asyncTaskCanceled",
               InvokeAsyncTaskFnWithId<&Agent::AsyncTaskCanceled>);
```

I assume the functions will be instantiated automatically as this kind of pattern is also used in [node_buffer.cc](https://github.com/nodejs/node/blob/master/src/node_buffer.cc?utf8=%E2%9C%93#L1232-L1244). I'm not too good at writing C++ though, so please excuse me if this is something impossible to do.